Boolean first edition

trane opened the pull request:
This is incomplete, but I want to get this up early for review
to make sure it is a useful level of abstraction and discussion

There are several things we can talk about here:
- Hoare logic
- Decision trees

Obviously, we need actual code here.

Ref: #4

trane committed:
First draft of boolean

This is incomplete, but I want to get this up early for review
to make sure it is a useful level of abstraction and discussion

There are several things we can talk about here:
 - Hoare logic
 - Decision trees

Obviously, we need actual code here.

Ref: #4

robinske commented on introduction/boolean.md:
```diff
@@ -0,0 +1,21 @@
+## Boolean
+
+Boolean means True or False. Would it surprise you to find out that the fundamentals of computation are built entirely out of True and False? We often call this concept "binary" -- meaning two choices. You'll often here these two names, "binary" and "boolean" used to describe the same concept.
+
```
here => hear

robinske commented on introduction/boolean.md:
```diff
@@ -0,0 +1,21 @@
+
+One of the primary things a programmer does everyday is figuring out how to take a human concept and turn it into something that the computer can run. Being such a fundamental building block of programming means that really complex processes can be broken down into chunks of what we can call "boolean logic".
+
+Our REPL uses boolean logic.
```
If we haven't already - let's add something like `REPL (which stands for Read, Evaluate, Print Loop)`

robinske commented on introduction/boolean.md:
```diff
@@ -0,0 +1,21 @@
+
+Our REPL uses boolean logic.
+
+The Read phase is boolean: "has the user hit Enter/Return yet"? If the answer is yes/true, then it takes whatever text was input before and sends that input to the Evaluate phase.
```
Have we introduced the REPL yet? How to access it? Otherwise we might need more context.

trane committed:
Add concrete examples of boolean operations

This adds some playground type stuff for the REPL and sticks it
at the beginning of this section, leaving the larger discussion for
after students have time to play around with the REPL.

Ref: #4

trane committed:
Fix typo in boolean

trane committed:
Add some context and more examples

Boolean logic is so amazing that I can barely controll my excitement.
I hope to pursuade students to explore the things they learn by
playing in the REPL and trying to answer their own questions by trying
it out.

Ref: #4
